stonesoup/simulator/transition.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+from datetime import timedelta
+from copy import deepcopy
+
+import numpy as np
+
+from stonesoup.models.transition.linear import ConstantTurn, ConstantVelocity,\
+    CombinedLinearGaussianTransitionModel
+from stonesoup.types.array import StateVector
+
+
+def create_smooth_transition_models(initial_state, x_coords, y_coords, times, turn_rate):
+    """Generate a list of constant-turn and constant acceleration transition models alongside a
+    list of transition times to provide smooth transitions between 2D cartesian coordinates and
+    time pairs.
+    An assumption is that the initial_state's x, y coordinates are the first elements of x_ccords
+    and y_coords respectively. Ie. The platform starts at the first coordinates.
+
+    Parameters
+    ----------
+    initial_state: :class:`~.State` The initial state of the platform.
+    x_coords:
+        A list of int/float x-coordinates (cartesian) in the order that the target must follow.
+    y_coords:
+        A list of int/float y-coordinates (cartesian) in the order that the target must follow.
+    times:
+        A list of :class:`~.datetime.datetime` dictating the times at which the target must be at
+        each corresponding coordinate.
+    turn_rate: Float
+        Angular turn rate (radians/second) measured anti-clockwise from positive x-axis.
+
+    Returns
+    -------
+    transition_models:
+        A list of :class:`~.ConstantTurn` and :class:`~.Point2PointConstantAcceleration` transition
+        models.
+    transition_times:
+        A list of :class:`~.datetime.timedelta` dictating the transition time for each
+        corresponding transition model in transition_models.
+
+    Notes
+    -----
+    x_coords, y_coords and times must be of same length.
+    """
+
+    state = deepcopy(initial_state)  # don't alter platform state with calculations
+
+    if not len(x_coords) == len(y_coords) == len(times):
+        raise ValueError('x_coords, y_coords and times must be same length')
+
+    transition_models = []
+    transition_times = []
+
+    for x_coord, y_coord, time in zip(x_coords[1:], y_coords[1:], times[1:]):
+
+        dx = x_coord - state.state_vector[0]  # distance to next x-coord
```

Comment:
It is assumed here (and several other places below) that `len(state.statev_vector)==6`, assuming [x, vx, y, vy, z, vz]. 

Is this safe? Could you get access to a position/velocity mapping to use instead? I'm not thinking that you add one to this class, but `Platform`s have them already.

Reply:
It is quite an assumption. Probably not necessary. I'll look in to making it less specific.

Reply:
This can be done in 2D cartesian too, so the model maker does not assume a 'z-direction'. While it is an assumption that the state should be of form (x, vx, y, vy), this form is contained completely within the model maker. It only returns transition models, which can then be used to apply to different state space arrangements according to how one sets up their own platforms (and corresponding position/velocity mappings).

Reply:
Perhaps it would be worth adding this information to the documentation